### ISSUE TITLE
paramiko trying to subscript a dictionary?

### ISSUE THREAD

**Issue by benson-basis** [paramiko/paramiko#703]
Using version paramiko (1.16.0):

```
  File "/usr/local/lib/python3.5/site-packages/paramiko/client.py", line 311, in connect
    raise NoValidConnectionsError(errors)
  File "/usr/local/lib/python3.5/site-packages/paramiko/ssh_exception.py", line 166, in __init__
    body = ', '.join([x[0] for x in addrs[:-1]])
TypeError: 'dict_keys' object is not subscriptable
```

**Comment by MarechJ** [comment]
Hi, I found the same bug. 

There's an easy fix, I'll try to submit a PR.

**Comment by halfnibble** [comment]
Do you have that easy fix? I'm needing to use this. Thanks.

**Comment by bitprophet** [comment]
FTR this sort of thing is a Python 3 specific issue, Python 3 changed many common dict and list methods to return special objects, so to function under 2+3 one needs to add more code :(

It was fixed in #679 and will be out in 1.16.1 within a day or two tops. Thanks!
